src/components/RankingSection.tsx

```diff
@@ -140,10 +146,28 @@ const ToggleButton = styled.button(({ theme }) => ({
 }));
 
 const RankingSection = () => {
-  const [selectedFilter, setSelectedFilter] = useState('전체');
-  const [selectedTab, setSelectedTab] = useState('받고 싶어한');
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  // 기본값 설정
+  const rawFilter = searchParams.get('filter');
+  const rawTab = searchParams.get('tab');
+  const rawExpanded = searchParams.get('expanded');
+
+  const selectedFilter: FilterType = FILTERS.includes(rawFilter as FilterType)
+    ? (rawFilter as FilterType)
+    : '전체';
+
+  const selectedTab: TabType = TABS.includes(rawTab as TabType)
+    ? (rawTab as TabType)
+    : '받고 싶어한';
+
+  const isExpanded = rawExpanded === 'true';
 
-  const [isExpanded, setIsExpanded] = useState(false);
+  const updateParam = (key: string, value: string) => {
+    const newParams = new URLSearchParams(searchParams.toString());
+    newParams.set(key, value);
+    setSearchParams(newParams);
+  };
```

Comment:
[제안]
setState가 이전 상태를 callback 형태로 받을 수 있다는 점을 고려하면 아래와 같이 더 간단하게 구현해볼 수도 있겠네요!
(https://reactrouter.com/api/hooks/useSearchParams#setsearchparams-function)

```suggestion
  const updateParam = (key: string, value: string) => {
     setSearchParams(searchParams => {
         searchParams.set(key,value);
         return searchParams;
     });
  };
```